### ISSUE TITLE
tf.image.convert_image_dtype(image, tf.float32) does not normalize output properly

### ISSUE THREAD

**Issue by Luonic** [tensorflow/tensorflow#19691]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary via pip
- **TensorFlow version (use command below)**: v1.8.0-0-g93bc2e2072 1.8.0
- **Python version**: 3.5.2
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

### Describe the problem
In docs: `This op converts between data types, scaling the values appropriately before casting.`
Yes, we can not scale uint8 and then cast, we will loose all the information
But this function is counterintuitive in a way that we can convert from float to uint and get ready-to-display result (e.g. in Tensorboard) but we can't do same to covert when reading images as input to our model
Is that a correct behaviour?

### Source code / logs
```
image_gt = tf.image.decode_image(image_parsed["image"], channels=3)
image_gt = tf.image.convert_image_dtype(image_gt, tf.float32, saturate=False)
image_gt = tf.Print(image_gt, [tf.reduce_max(image_gt)], "image_gt_cast")
```

Result of execution:
```
image_gt_cast[250.892319]
image_gt_cast[244.243454]
image_gt_cast[245.468872]
image_gt_cast[258.527466]
image_gt_cast[270.992615]
image_gt_cast[227.410767]
```

**Comment by facaiy** [comment]
Could you please explain what result your expect in your example?

**Comment by Luonic** [comment]
@facaiy yes I am expecting scaling of values as written in the docs
```
Images that are represented using floating point values are expected to have values in the range [0,1). Image data stored in integer data types are expected to have values in the range [0,MAX], where MAX is the largest positive representable number for the data type.

This op converts between data types, scaling the values appropriately before casting.
```
I looked at the source of this function and there is some scaling-related code, but I don't shure that it works at all. Is this intended behavior of this OP? As it looks to me it allows at best only to get rid of `tf.cast(to.clip_by_value(image, 0.0, 1.1), tf.uint8)` when converting to integers with `saturate=True`
BTW converting from float to uint8 gave me black images (0, 1) -> (0,255) without scaling result in all zeros.
And there in some inconsistency between tensorboard's image float range and this function (and probably all funcs in tf.image, tensorboard's float images are in (-1,1) range and tf.image works with (0,1).

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 89 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tatianashp** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by facaiy** [comment]
@drpngx Could you comment or reassign?

**Comment by tatianashp** [comment]
Reopening since there's still active interest.

**Comment by tatianashp** [comment]
@facaiy My apologies for letting this fall through the cracks. 

@drpngx, @jart Can you help here?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by drpngx** [comment]
Sorry for the delay. @Luonic I don't understand what the problem is. Could you elaborate maybe? Give a small numerical example?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by drpngx** [comment]
Can you print `image_gt.dtype` and `image_gt.dtype.max`?

Here's what I did:
```
x = tf.constant([1 2 255], dtype=tf.uint8)
sess.run(tf.convert_image_dtype(x, dtype=tf.float32))
```
and I'm getting (as expected):
```array([0.00392157, 0.00784314, 1.        ], dtype=float32)```

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 44 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
We are closing this issue for now due to lack of activity. Please comment if this is still an issue for you. Thanks!
